evil-states.el

```diff
@@ -881,7 +881,7 @@ CORNER defaults to `upper-left'."
          (this-command #'evil-replace-backspace))
     (when prev
       (delete-char 1)
-      (when char (save-excursion (insert char))))))
+      (when char (save-excursion (insert-and-inherit char))))))
```

Comment:
As mentioned, I think this is one function that shouldn't inherit when inserting.